mcp/transport.go

```diff
@@ -103,6 +103,29 @@ func NewStdioTransport() *StdioTransport {
 	return &StdioTransport{}
 }
 
+// An IOTransport is a [Transport] that communicates over separate
+// io.ReadCloser and io.WriteCloser using newline-delimited JSON.
+type IOTransport struct {
+	reader io.ReadCloser
```

Comment:
Let's just make these exported fields, so that the transport is extensible, and remove NewIOTransport.

See also #286